File: Loenn/entities/bloomPoint.lua
local utils = require("utils")
local drawableSprite = require("structs.drawable_sprite")
local bloomSprite = require("mods").requireFromPlugin("libraries.bloomSprite")

local bloomPoint = {
    name = "FrostHelper/BloomPoint",
    placements = {
        name = "default",
        data = {
            alpha = 1.0,
            radius = 16.0,
        }
    },
}

function bloomPoint.sprite(room, entity)
    return bloomSprite.getSprite(entity, entity.alpha, entity.radius)
end

function bloomPoint.selection(room, entity)
    local radius = entity.radius
    local halfRadius = radius / 2


    local main = utils.rectangle(entity.x - halfRadius, entity.y - halfRadius, radius, radius)

    return main
end



return bloomPoint

File: Code/FrostHelper/Entities/BloomPoint.cs
namespace FrostHelper;

[CustomEntity("FrostHelper/BloomPoint")]
public class BloomPointEntity : Entity {
    public BloomPointEntity(EntityData data, Vector2 offset) : base(data.Position + offset) {
        Add(new BloomPoint(data.Float("alpha", 1f), data.Float("radius", 16f)));
    }
}
